Remove sneaky @reach/router usage

echocrow opened the pull request:
Seems like a `"@reach/router"` import snug in or was missed when resolving merge from #800 to #802.

I've kept the usage of `location.state?`, though based on a quick code search, I don't see this being set or used anywhere else. May be stale code and safe to remove?

kasperkristensen reviewed (approved):
LGTM! You can remove the `useLocation`, its dead code from when duplicating a discount would open up the create page rather than just copying it immediately 😃

echocrow commented:
> LGTM! You can remove the `useLocation`, its dead code from when duplicating a discount would open up the create page rather than just copying it immediately 😃

removed!

also deleted more unused discount code while I was at it (couple props that were never used or set). lmk if you'd prefer to keep those "just in case", and I'll undo the extra trimming
